From a79dbd81872484b3f5f14dc759464759ecd1eccc Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Thu, 05 Jun 2025 11:12:19 +0300
Subject: [PATCH] [SW-230480] i2c: skip read sec. reg. for winbond flash

Skip read security register for Winbond flash
during OOB recovery via I2C. For Winbond flash
this command is not supported. Instead, it has
3 256 byte security registers, that can be filled
with any OEM specific information and we are not
using them

Change-Id: I21a9f21403ac34d2e6056a04fb57d82eb8560dda
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/src/spi-cpld.cpp b/src/spi-cpld.cpp
index 8358150..6a5d8b5 100644
--- a/src/spi-cpld.cpp
+++ b/src/spi-cpld.cpp
@@ -111,11 +111,13 @@
 static int check_sec()
 {
 	int err;
-	__u8 sec;
+	__u8 sec = 0;
 
-	err = spi_dev_special_cmd(CMD_READ_SECURITY, &sec);
-	if (err)
-		return err;
+	if (IS_MACRONIX(cfi_mfr_id)) {
+		err = spi_dev_special_cmd(CMD_READ_SECURITY, &sec);
+		if (err)
+			return err;
+	}
 
 	return sec;
 }
